test/corelogging.jl

```diff
@@ -114,6 +114,35 @@ end
         @test only(collect_test_logs(logmsg)[1]).kwargs[:x] === "the y"
     end
 end
+@testset "Log message handle_message exception handling" begin
+    function capture_stderr(func)
+        fname = tempname()
+        f = open(fname, "w")
+        redirect_stderr(f) do
+            func()
+        end
+        close(f)
+        buf = read(fname)
+        rm(fname)
+        String(buf)
+    end
+
+    # Exceptions in log handling (printing) of msg are caught by default
+    struct Foo end
+    Base.show(::IO, ::Foo) = 1 ÷ 0
+    out = capture_stderr() do
+        @info Foo()
+    end
+    @test occursin("Error: Exception while generating log record in module Main at", out)
+    @test occursin("DivideError: integer division error", out)
```

Comment:
It is preferable to use existing builtin handling, rather than making custom implementations
```suggestion
    # Exceptions in log handling (printing) of msg are caught by default
    struct Foo end
    Base.show(::IO, ::Foo) = 1 ÷ 0

    @test_warn r"Error: Exception while generating log record in module Main at.*DivideError: integer division error" @info Foo()
```

Although looks like this is supposed to be logging test (`@test_logs`), not an IO test (`@test_warn`)?

Reply:
Ah, thanks, i didn't know about `@test_warn`! Fantastic.

Yes, I started out with `@test_logs`, however I ran into the issue that `@test_logs` doesn't actually call `handle_message` on the log messages, it just returns them as-is, so we do not catch the exceptions I expected. So we need to test the stderr content directly, I believe.

Reply:
I will improve the comment to explain this better.

Reply:
Done! Can you give this one more look? :) Thanks!